java/kotlin-extractor/src/main/kotlin/KotlinFileExtractor.kt

```diff
@@ -4101,6 +4451,163 @@ open class KotlinFileExtractor(
         extractExpressionExpr(loop.condition, callable, id, 0, id)
     }
 
+    @Suppress("UNCHECKED_CAST")
+    private fun <T : DbExpr> exprIdOrFresh(id: Label<out DbExpr>?) = id as? Label<T> ?: tw.getFreshIdLabel()
```

Comment:
I think we can avoid the warning suppression here by using something like `id?.cast() ?: tw.getFreshIdLabel()`